### PR TITLE
Codeintel: Make index usable

### PR DESCRIPTION
**Pull request by camdencheek**
We were running into a performance issue where a query that did `ORDER BY last_removal_time DESC, document_id` could not take advantage of an index on `last_removal_time, document_id` because the secondary order does not match.

This replaces the unusable index with an index on `last_removal_time DESC, document_id` so that the query can depend on it rather than sorting millions of rows.

## Test plan

Manually ran a modified query on sourcegraph.com that confirmed that an `ORDER BY` that can take advantage of the index reduces runtime of the query from ~7500ms to ~30ms. Since the index is not currently being used, the risk here is low -- at worst this has no effect.

### CODE REVIEW

File: migrations/codeintel/1678898749_make_unreferenced_documents_index_usable/up.sql
Comment (camdencheek) [inline migrations/codeintel/1678898749_make_unreferenced_documents_index_usable/up.sql:1]: Migration 1: make a new index on `last_removal_time DESC, document_id`

File: migrations/codeintel/1678899132_remove_unused_unreferenced_documents_index/up.sql
Comment (camdencheek) [inline migrations/codeintel/1678899132_remove_unused_unreferenced_documents_index/up.sql:1]: Migration 2: remove the old index that is unused.

File: internal/gitserver/v1/gitserver.pb.go
Comment (camdencheek) [inline internal/gitserver/v1/gitserver.pb.go:1]: This is unrelated, but was output by `sg generate`. As a followup, we should probably pin these versions.

File: internal/gitserver/v1/gitserver.pb.go [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
-// 	protoc-gen-go v1.29.0
+// 	protoc-gen-go v1.29.1
```
Comment (camdencheek) [inline internal/gitserver/v1/gitserver.pb.go:3]: This is unrelated, but was output by `sg generate`. As a followup, we should pin these versions.

**Review by efritz — APPROVED** [review]

**Pull request merged**